charts/seed-monitoring/charts/core/charts/prometheus/rules/node-exporter.rules.yaml

```diff
@@ -1,6 +1,17 @@
 groups:
 - name: node-exporter.rules
   rules:
+  - alert: NodeKubeletDown
+    expr: count_over_time((sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]) == 9
```

Comment:
Had a combined look with @ScheererJ and @shafeeqes. We have following concerns:
- Query for Prometheus and Grafana is not the same.
- Does the node get deleted/removed if it is in `NotReady` Condition for 10 min, In that case we should reduce the timing in query to 9 or 8 minutes.
- Should the dashboard be part of the `node-exporter`,  as of now it comes as a separate block that is not properly aligned.
- Node count scale can't have a decimal value.
<img width="1651" alt="Screenshot 2023-03-21 at 3 52 35 PM" src="https://user-images.githubusercontent.com/28653770/226578269-8035a6dd-b433-4668-9cdf-8e0fa3ded0a8.png">

/cc @istvanballok @rickardsjp

Reply:
To address some of the aforementioned concerns, we have conducted several rounds of testing. The reason behind setting the threshold at 10 minutes is to minimize the occurrence of false positives. Our testing has shown that a variety of factors can trigger a node to be deemed "not ready," including updates to k8s version, GardenLinux, and patch versions.

In our analysis, we have considered a worst-case scenario, assuming an empty cluster. In such a case, any event lasting longer than 10 minutes may be deemed meaningful, particularly if it persists for an extended period. Our objective is to filter out non-essential events and only flag nodes that are potentially experiencing genuine issues.

Not sure if that dashboard should be elsewhere. I'm open to the discussion. 

The node count scale will adapt when values are generated. 

---


The explanation of the expression : 

Breaking down the PromQL expression:
The expression is:
expr: count_over_time((sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]) == 10

kube_node_status_condition{condition="Ready",status!="true"} == 1: prints a table of nodes where the status is not true (printed are nodes with status false or unknown) (== 1 means that the lines on the table satisfy this expression)

((kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]: For EACH data point, performs the above query every evaluation interval for the past 10 minute window. Here, we have evaluation_interval: 1m, so this would check once every minute for the past 10 minutes. Because == 1 exists, this will only show lines where this expression was true. So if we have 10 elements, the condition was true for 10 minutes.

(sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]: (sum by (node) groups the results by node.

count_over_time((sum by (node)(kube_node_status_condition{condition="Ready",status!="true"}) == 1)[10m:]) == 10: count_over_time counts how many times the condition is true (instead of a 1 at every evaluation interval, this counts how many times the query returns 1). The == 10 part only considers nodes that have had this condition true for every minute in the past 10 minute evaluation time.

Reply:
Please also note that the amount of time until a non-ready node gets replaced is configurable. Therefore, a static check like this might always have false positives if a cluster owner choses a different value, e.g. 20 minutes.